src/translator/text-based-field.ts

```diff
@@ -66,6 +76,43 @@ export namespace TextBasedField {
         };
     }
 
+    const createTextFieldValidationSchema = (validation: IValidation[]) => {
+        const validationObj = validation.reduce((acc, value) => {
```

Comment:
declare typing with generics instead of casting

```suggestion
        const validationObj = validation.reduce<ISchemaValidation[]>((acc, value) => {
```